pygmt/src/triangulate.py

```diff
@@ -0,0 +1,116 @@
+"""
+triangulate - Delaunay triangulation or Voronoi partitioning and gridding of
+Cartesian data.
+"""
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    I="spacing",
+    J="projection",
+    R="region",
+    V="verbose",
+    b="binary",
+    d="nodata",
+    e="find",
+    f="coltypes",
+    h="header",
+    i="incols",
+    r="registration",
+)
+@kwargs_to_strings(R="sequence")
+def triangulate(table=None, x=None, y=None, z=None, **kwargs):
```

Comment:
Wait for #1479 to decide on whether to use parameter name 'table' or 'data'.

Reply:
Renamed parameter 'table' to 'data' in d00d34756b1fb2b9e704c0608af3960264624563.